buildconfig/CMake/CommonSetup.cmake

```diff
@@ -258,20 +258,33 @@ endif ()
 ###########################################################################
 
 if ( CMAKE_VERSION VERSION_GREATER "3.5" )
+  set(DEFAULT_CLANG_TIDY_CHECKS "-*,performance-for-range-copy,performance-unnecessary-copy-initialization,modernize-use-override,modernize-use-nullptr,modernize-loop-convert,modernize-use-bool-literals,modernize-deprecated-headers,misc-*,-misc-unused-parameters")
   set(ENABLE_CLANG_TIDY OFF CACHE BOOL "Add clang-tidy automatically to builds")
   if (ENABLE_CLANG_TIDY)
     find_program (CLANG_TIDY_EXE NAMES "clang-tidy" PATHS /usr/local/opt/llvm/bin )
     if (CLANG_TIDY_EXE)
+      set(APPLY_CLANG_TIDY_FIX OFF CACHE BOOL "Apply fixes found through clang-tidy checks")
```

Comment:
Could we update the `set` functions that provide CACHE variables to use the more modern [`option`](https://cmake.org/cmake/help/v3.13/command/option.html) command? It signals the intent much better I think.